From 30e808f2ff39ce487e5348419fd372a895430275 Mon Sep 17 00:00:00 2001
From: Boris Brezillon <boris.brezillon@collabora.com>
Date: Fri, 15 Sep 2023 17:17:14 +0200
Subject: [PATCH 46/64] panfrost: Inline panfrost_emit_heap_set()

Only called once from the panfrost_csf_init_context() function
---
 src/gallium/drivers/panfrost/pan_cmdstream.c | 21 +++++++-------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/src/gallium/drivers/panfrost/pan_cmdstream.c b/src/gallium/drivers/panfrost/pan_cmdstream.c
index dbf3c1c0a82..749c98f75e7 100644
--- a/src/gallium/drivers/panfrost/pan_cmdstream.c
+++ b/src/gallium/drivers/panfrost/pan_cmdstream.c
@@ -2662,19 +2662,6 @@ panfrost_initialize_surface(struct panfrost_batch *batch,
    }
 }
 
-static void
-panfrost_emit_heap_set(struct panfrost_batch *batch, u64 heap_ctx_gpu_va)
-{
-#if PAN_USE_CSF
-   ceu_builder *b = batch->ceu_builder;
-
-   /* Setup the tiler heap */
-   ceu_index heap = ceu_reg64(b, 72);
-   ceu_move64_to(b, heap, heap_ctx_gpu_va);
-   ceu_heap_set(b, heap);
-#endif
-}
-
 static void
 panfrost_emit_batch_end(struct panfrost_batch *batch)
 {
@@ -5061,7 +5048,13 @@ panfrost_csf_init_context(struct panfrost_context *ctx)
    }
 
    struct panfrost_batch *batch = panfrost_get_batch_for_fbo(ctx);
-   panfrost_emit_heap_set(batch, thc.tiler_heap_ctx_gpu_va);
+
+   /* Setup the tiler heap */
+   ceu_builder *b = batch->ceu_builder;
+   ceu_index heap = ceu_reg64(b, 72);
+   ceu_move64_to(b, heap, thc.tiler_heap_ctx_gpu_va);
+   ceu_heap_set(b, heap);
+
    batch->any_compute = 1;
    panfrost_flush_all_batches(ctx, "Gfx queue init");
 }
-- 
2.42.0

